src/index.ts

```diff
@@ -340,7 +340,24 @@ export function HTPipePreAuthorize<
           ...leftContextOut,
         };
         const rightOut = right.preAuthorize(rightIn);
```

Comment:
Simpler:
1. early-return false if rightPassed isn't true
2. early-return true if it's the special-case that both leftOut and rightOut are strictly true
3. otherwise, it's the normal case of returning {...leftContextOut, rightContextOut}

- [ ] simplify